apps/api/src/jobs/consumers/notifications/notification.consumer.ts

```diff
@@ -25,15 +25,51 @@ export abstract class NotificationConsumer {
     try {
       this.logger.log(`Sending notification with id: ${id}`);
       const result = await sendNotification();
-      notification.deliveryStatus = DeliveryStatus.SUCCESS;
+      notification.deliveryStatus = DeliveryStatus.AWAITING_CONFIRMATION;
       notification.result = { result };
     } catch (error) {
-      notification.deliveryStatus = DeliveryStatus.FAILED;
+      notification.deliveryStatus = DeliveryStatus.PENDING;
+      notification.retryCount++;
       notification.result = { result: error };
       this.logger.error(`Error sending notification with id: ${id}`);
       this.logger.error(JSON.stringify(error, ['message', 'stack'], 2));
     } finally {
       await this.notificationRepository.save(notification);
     }
   }
+
+  async processAwaitingConfirmationNotificationQueue(
+    job: Job<number>,
+    getNotificationStatus: () => Promise<{
+      result: unknown;
+      deliveryStatus: number;
+    }>,
+  ): Promise<void> {
+    const id = job.data;
```

Comment:
### Refactoring Needed

1. **Separate Queues**:
   - Avoid using the same queues for sending notifications and fetching status updates.
   
2. **Retry Mechanism**:
   - Use the same queue for retries as for sending notifications.
   - Retry notifications with `PENDING` status and increment the retry count.
   - At failure states, update the status to `PENDING` (for retry) or `FAILED` (no retry).

3. **Cron Job for Status Updates**:
   - Fetch notifications with `AWAITING CONFIRMATION` status.
   - Push them to separate queues per provider type for fetching status updates.
   - Update status based on the response: `PENDING` for retry, `FAILED` or `SUCCESS` as appropriate.
   - Include conditional checks to determine failure or success.
   - If status isn't achieved during the call, skip it for the next cron run.
   - When pushing to the queue, consider changing the status to `QUEUED`.

### Note

- Ideally, use a queue per provider, not just per provider type.
- Example: If there are two providers for different apps, handle them in separate queues. One provider might send many notifications while another sends few. This approach allows for parallel processing and avoids sequential execution.
- Can create at ask for this for now.

Reply:
These issues are addressed as part of the recently merged https://github.com/OsmosysSoftware/osmo-x/pull/264 and in the commit https://github.com/OsmosysSoftware/osmo-x/pull/263/commits/c9eca985bed83d84681bb43346b3ff44db998cba